biz.aQute.resolve/src/biz/aQute/resolve/Bndrun.java

```diff
@@ -185,4 +194,100 @@ public Collection<Container> getRunbundles() throws Exception {
 		return super.getRunbundles();
 	}
 
+	private Collection<Container> cache() throws Exception {
+		File ours = getPropertiesFile();
+		File cache = getCacheFile(ours);
+		File workspace = getWorkspace().getPropertiesFile();
+		if (workspace == null)
+			workspace = ours;
+
+		String force;
+		if (!cache.isFile())
+			force = "no cache file";
+		else if (cache.lastModified() <= ours.lastModified())
+			force = "cache file is older than our properties";
+		else if (cache.lastModified() <= workspace.lastModified())
+			force = "cache file is older than our workspace";
+		else
+			force = null;
+		trace("force = %s", force);
+
+		boolean tryCache = force == null;
+
+		try (Processor p = new Processor()) {
+
+			if (cache.isFile())
+				p.setProperties(cache);
+
+			if (tryCache) {
+				trace("attempting to use cache");
+				String runbundles = p.getProperty(Constants.RUNBUNDLES);
+				Collection<Container> containers = parseRunbundles(runbundles);
+				if (isAllOk(containers)) {
+					return containers;
+				} else {
+					trace("the cached bundles were not ok, will resolve");
+				}
+			}
+			trace("resolving");
+
+			IO.delete(cache);
+			if (cache.isFile()) {
+				throw new IllegalStateException("cannot delete cache file " + cache);
+			}
+
+			RunResolution resolved = RunResolution.resolve(this, Collections.emptyList());
+			if (!resolved.isOK()) {
+				throw new IllegalStateException(resolved.report(false));
+			}
+
+			String spec = resolved.getRunBundlesAsString();
+
+			List<Container> containers = parseRunbundles(spec);
+			if (isAllOk(containers)) {
+				UTF8Properties props = new UTF8Properties(p.getProperties());
+				props.setProperty(Constants.RUNBUNDLES, spec);
+				cache.getParentFile()
+					.mkdirs();
+				props.store(cache);
+			}
+			return containers;
+		}
+	}
+
+	/**
+	 * Return the file used to cache the resolved solution for the given file
+	 *
+	 * @param file the file to find the cached file for
+	 * @return the cached file
+	 */
+	public File getCacheFile(File file) {
+		try {
+			String path = file.getAbsolutePath();
+			String digest = SHA1.digest(path.getBytes(StandardCharsets.UTF_8))
+				.asHex();
+			return getWorkspace().getCache("resolved-cache/".concat(digest)
+				.concat(".resolved"));
+		} catch (Exception e) {
+			// not gonna happen
+			throw Exceptions.duck(e);
+		}
+	}
+
+	boolean testIgnoreDownloadErrors = false;
```

Comment:
Why does testing need this? Seems unusual that the test does not test the normal code path.

Reply:
Because we then need all dependencies ... that is either potential communication errors. That said, I changed the test from the enroute set to a much smaller set so I can see if I can have them locally.